### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Tracy-miranja**
- [ ] Set up React Testing Library.
- [ ] Wrote unit tests for the files operate.js and calculate.js using Jest.
- [ ] Created unit tests for all React components:
- [ ] Used Jest snapshots to test the components.
- [ ] Used React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/components/test/Calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (divinecharlotte) [inline src/components/test/Calculator.test.js:1]: - [ ] kindly consider making sure that all your tests are passing  as they should to full-fill the project requirement cause currently all your tests are failing

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @ Team       , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 


- Descriptive commit message  ✔️
- linters are passing 👍  

## Required Changes ♻️

- [ ] kindly consider adding a working Linkdhin link for Tracy miranja  you added in the authors section and also consider including all team members in this section to make it look more professional and it list all contributors 

- [ ] kindly consider making sure that you add a full description in your readme on how someone can have a code of your project and how to interact with it this will avoid confusion to the people visiting your project and want to have a copy

_Check the comments under the review._



## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/test/Calculator.test.js (hunk 1 above)
Comment (Tracy-miranja) [reply src/components/test/Calculator.test.js:1]: Hi @divinecharlotte i don't understand this, all the test are passing on my end, i don't get it when you say they are all failing

Comment (Tracy-miranja) [reply src/components/test/Calculator.test.js:1]: please recheck

Comment (divinecharlotte) [reply src/components/test/Calculator.test.js:1]: hey @Tracy-miranja  here is the image below 👇 
Screen Shot 2023-04-06 at 23 31 53

Comment (iambenkis) [inline src/components/test/Calculator.test.js:1]: - [ ] As requested by the previous Code reviewer, make sure all the testes are passing 😄 , most of them are passing but not all 👍 
Capture

**Review by iambenkis — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Tracy-miranja   
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 

- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @iambenkis  when you comment, so I can receive the notification or use Slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/test/Calculator.test.js (hunk 1 above)
Comment (Tracy-miranja) [reply src/components/test/Calculator.test.js:1]: hello i think what you need to do is install the react-test-render

Comment (Tracy-miranja) [reply src/components/test/Calculator.test.js:1]: can i have a meeting with you please

Comment (Tracy-miranja) [reply src/components/test/Calculator.test.js:1]: coz i see all the project are passing on my end, even on my partners end

**Review by codecaiine — APPROVED** [review]
Hi @Tracy-miranja , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great changes requested application ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tracy-miranja** [comment]
Thanks for the feedback @codecaiine

**Pull request merged**